test/smoke/src/sql/areas/notebook/notebook.test.ts

```diff
@@ -146,70 +146,94 @@ export function setup(opts: minimist.ParsedArgs) {
 		});
 
 		describe('Cell Toolbar Actions', function () {
-			const sampleText: string = 'Test Text';
-			async function createCellAndSelectAllText(app: Application): Promise<void> {
+			async function verifyCellToolbarBehavior(app: Application, toolbarAction: () => Promise<void>, selector: string, checkIfGone: boolean = false): Promise<void> {
+				const sampleText: string = 'Test Text';
+
 				await app.workbench.sqlNotebook.newUntitledNotebook();
 				await app.workbench.sqlNotebook.addCellFromPlaceholder('Markdown');
 				await app.workbench.sqlNotebook.waitForPlaceholderGone();
 				await app.workbench.sqlNotebook.textCellToolbar.changeTextCellView('Split View');
 				await app.workbench.sqlNotebook.waitForTypeInEditor(sampleText);
 				await app.workbench.sqlNotebook.selectAllTextInEditor();
+
+				await toolbarAction();
+				await app.code.dispatchKeybinding('escape');
+				if (checkIfGone) {
+					await app.workbench.sqlNotebook.waitForTextCellPreviewContentGone(selector);
+				} else {
+					await app.workbench.sqlNotebook.waitForTextCellPreviewContent(sampleText, selector);
+				}
 			}
 
 			it('can bold selected text', async function () {
 				const app = this.app as Application;
-				await createCellAndSelectAllText(app);
-				await app.workbench.sqlNotebook.textCellToolbar.boldSelectedText();
-				await app.code.dispatchKeybinding('escape');
-				await app.workbench.sqlNotebook.waitForTextCellPreviewContent(sampleText, 'p strong');
+				await verifyCellToolbarBehavior(app, () => app.workbench.sqlNotebook.textCellToolbar.boldSelectedText(), 'p strong');
+			});
+
+			it('can undo bold text', async function () {
+				const app = this.app as Application;
+				let boldText = () => app.workbench.sqlNotebook.textCellToolbar.boldSelectedText();
+				await verifyCellToolbarBehavior(app, () => boldText().then(() => boldText()), 'p strong', true);
```

Comment:
nit, but i would find this easier to just use async 

```
async () => {
   await boldText();
   await boldText();
}
```

Reply:
I only used a then() to keep it on the same line. I'll switch over to awaits.